File: backend/spec/factories/open_call.rb
FactoryBot.define do
  factory :open_call do
    investor

    sequence(:name) do |n|
      "Open call #{n}"
    end
    status { "launched" }

    picture { Rack::Test::UploadedFile.new(Rails.root.join("spec/fixtures/files/picture.jpg"), "image/jpeg") }

    country
    municipality
    department

    sdgs { [1, 4, 5] }
    instrument_types { ["loan"] }

    trusted { false }

    maximum_funding_per_project { 100_000 }

    sequence(:description) do |n|
      Faker::Config.random = Random.new(n)
      Faker::Lorem.paragraph(sentence_count: 4)
    end
    sequence(:funding_priorities) do |n|
      Faker::Config.random = Random.new(n)
      Faker::Lorem.paragraph(sentence_count: 4)
    end
    sequence(:funding_exclusions) do |n|
      Faker::Config.random = Random.new(n)
      Faker::Lorem.paragraph(sentence_count: 4)
    end
    sequence(:impact_description) do |n|
      Faker::Config.random = Random.new(n)
      Faker::Lorem.paragraph(sentence_count: 4)
    end

    language { "en" }

    closing_at { 10.months.from_now }
  end
end

File: frontend/validations/open-call.ts
import { yupResolver } from '@hookform/resolvers/yup';

import openCallSchemma from 'schemas/open-call';
import { OpenCallForm } from 'types/open-calls';

export default (section: number) => {
  return yupResolver(openCallSchemma(section));
};

export const formPageInputs: Partial<keyof OpenCallForm>[][] = [
  ['name', 'picture', 'country_id', 'department_id', 'municipality_id', 'description'],
  ['impact_description', 'sdgs'],
  ['maximum_funding_per_project', 'instrument_types', 'funding_priorities', 'funding_exclusions'],
  ['closing_at'],
];
